### PR TITLE
[front] - refacto(hooks): members hooks

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR refactors the members hooks in `front`, specifically moving the `useSearchMembers` hook from the `user.ts` file to the `memberships.ts` file. The changes aim to improve code organization and consistency by placing member-related hooks in a more appropriate location.

The main changes include:
- Moving the `useSearchMembers` hook from `lib/swr/user.ts` to `lib/swr/memberships.ts`.
- Updating import statements in affected components to use the new location of the `useSearchMembers` hook.
- Minor adjustments to the `useHandleMembersRoleChange` hook for better consistency and reusability.

## Risk

Low. The refactoring primarily involves moving existing code to a more appropriate location.

## Deploy Plan

Deploy `front`

### CODE REVIEW

File: front/hooks/useHandleMembersRoleChange.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import { useSWRConfig } from "swr";
+
+import { SendNotificationsContext } from "@app/components/sparkle/Notification";
+
+type HandleMembersRoleChangeParams = {
+  members: UserTypeWithWorkspaces[];
+  role: RoleType;
+};
+
+export function useHandleMembersRoleChange({
```
Comment (Fraggle) [inline front/hooks/useHandleMembersRoleChange.ts:16]: Thanks for creating the hook :)

Couple of nits:
- `useChangeMembersRoles()` instead ?
- Put the mutation(s) directly there when success ?
- Move at the end of "swr/memberships" (to keep CRUD actions together ?)

File: front/hooks/useHandleMembersRoleChange.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+          return false;
+        } else {
+          sendNotification({
+            type: "success",
+            title: "Role updated",
+            description: `Role updated to ${role} for ${members.length} member(s).`,
+          });
+
+          // Mutate the members data to trigger a revalidation
+          await mutate(`/api/w/${owner.sId}/members`);
```
Comment (Fraggle) [inline front/hooks/useHandleMembersRoleChange.ts:66]: Can you change with:

```
const {mutateMembers} = useMembers();
const {mutateMembersSearch} = useSearchMembers();
...

void mutateMembers();
void mutateMembersSearch();
```

That way you don't have the magical string and you mutate also the search (so you can remove it from the component).

Comment (Fraggle) [reply front/hooks/useHandleMembersRoleChange.ts:66]: Note we will automatically mutate all swr with the same base url (regardless of query params).

File: front/hooks/useHandleMembersRoleChange.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+          return true;
+        }
+      } catch (error) {
+        console.error("Error updating member roles:", error);
```
Comment (Fraggle) [inline front/hooks/useHandleMembersRoleChange.ts:70]: drop ?

**Review by Fraggle — APPROVED** [review]
Thansk for moving to a hook, I added a couple suggestions.

File: front/hooks/useChangeMembersRoles.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  role: RoleType;
+};
+
+export function useChangeMembersRoles({
+  owner,
+}: {
+  owner: LightWorkspaceType;
+}) {
+  const sendNotification = useContext(SendNotificationsContext);
+  const { mutateMembers } = useMembers({
```
Comment (Fraggle) [inline front/hooks/useChangeMembersRoles.ts:22]: don't forget to mutate search (a bug daph fixed this morning)

Comment (PopDaph) [reply front/hooks/useChangeMembersRoles.ts:22]: Yes please 🙏🏻 😄

**Pull request merged**